src/containers/NavBar.tsx

```diff
@@ -159,10 +166,41 @@ const Icon = styled.span`
   margin-right: 5px;
 `;
 
+const ThemeToggleContainer = styled.div`
+  display: flex;
+  align-items: center;
+
+  @media (max-width: 768px) {
+    position: absolute;
+    right: 10px;
+    top: 20px;
+    z-index: 10;
+  }
+`;
+
 // Functional component to render the Navbar - used on all pages
-export function NavBar({ links }: NavBarProps) {
+export function NavBar({ links, themeToggler, theme }: NavBarProps) {
   const location = useLocation();
   const isAuthenticated = localStorage.getItem("authenticated") === "true";
+  const themeContext = useTheme();
+  const storedTheme = (localStorage.getItem("theme") as ThemeType) || ThemeType.LIGHT;
+  const isDarkMode = storedTheme === ThemeType.DARK;
+  const [darkMode, setDarkMode] = useState(isDarkMode);
```

Comment:
I don't get it why do we need another state when we already have a state defined in the App.tsx